### ISSUE TITLE
Structure document lists correctly in reading mode

### ISSUE THREAD

**Issue by Ovgodd** [TypeCellOS/BlockNote#2809]
### What’s broken?

In reading mode, lists present in document content are not structured with `<ul>` and `<li> `elements. The list type is indicated by a data-content-type="bulletListItem" attribute on a `<div>`, without semantic value. Lists are rendered visually but are not properly conveyed by assistive technologies.

### What did you expect to happen?

Each list must be marked up with `<ul>` (unordered list) or `<ol>` (ordered list) and its items with `<li>`, in order to be correctly conveyed by assistive technologies.

Before:

```
<div class="bn-block" data-node-type="blockContainer">
  <div class="bn-block-content" data-content-type="bulletListItem">
    <p class="bn-inline-content">…</p>
  </div>
</div>

```
After:

```
<ul>
  <li><p>…</p></li>
</ul>
```

### Steps to reproduce

Open a document containing a list in Docs in reading mode (reader role).
Inspect the list's source code.
Observe the absence of `<ul> / <ol> and <li>` elements.
Possible solution

### BlockNote version

0.51.1

### Additional context

A blind user navigating with a screen reader cannot identify lists as such. They do not benefit from "list of X items" announcements and cannot navigate from item to item using list navigation shortcuts.

**Comment by YousefED** [comment]
Best would be to use ul / ol wrappers and li elements, but this would likely need a plugin to render decorations around groups of list items.

An alternative might be to use `role="list"` and `role="listitem"`, but even this would require a wrapper element around list siblings
